### PR TITLE
Include log file name to every line streamed out

### PR DESCRIPTION
**Pull request by CharityKathure**
image

### CODE REVIEW

File: LogMonitor/src/LogMonitor/LogFileMonitor.cpp [hunk 1]
```diff
@@ -1789,7 +1788,8 @@ LogFileMonitor::ReadLogFile(
                         //
                         // newLineBuffer was empty, so only print the found line.
                         //
-                        logWriter.WriteConsoleLog(decodedString.substr(0, found));
+                        //std::wstring decodedAll = Utility::ReplaceAll(decodedString.substr(0, found), L"\n", L"\n" + fileName);
```
Comment (profnandaa) [inline LogMonitor/src/LogMonitor/LogFileMonitor.cpp:1791]: nit: if no longer in use, can just delete the line.

Comment (CharityKathure) [reply LogMonitor/src/LogMonitor/LogFileMonitor.cpp]: This has been corrected. Thank you for pointing it out.

File: LogMonitor/LogMonitorTests/LogFileMonitorTests.cpp [hunk 2]
```diff
@@ -950,7 +950,7 @@ namespace LogMonitorTests
                 // It should the FULL content of the file, that's why we need
                 // to concatenate twice the content.
                 //
-                Assert::AreEqual((TO_WSTR(content) + TO_WSTR(content) + L"\n").c_str(), output.c_str());
+                Assert::IsTrue(output.find((TO_WSTR(content) + TO_WSTR(content) + L"\n").c_str()) != std::wstring::npos);
```
Comment (profnandaa) [inline LogMonitor/LogMonitorTests/LogFileMonitorTests.cpp:953]: why the change to `IsTrue`, do we still need the concatenation (+) then?

**Review by profnandaa — APPROVED** [review]
LGTM, just one commet.

**Review by bobsira — APPROVED** [review]

File: LogMonitor/LogMonitorTests/LogFileMonitorTests.cpp (hunk 2 above)
Comment (CharityKathure) [reply LogMonitor/LogMonitorTests/LogFileMonitorTests.cpp:953]: This is because the stdout output will not only contain the concatenation of the contents but also the file name at the beginning. 

'AreEqual' at the point will always return false. However, checking if the output contains the concatenation should be true if the 'WriteToFile' function worked correctly.


image

File: LogMonitor/src/LogMonitor/LogFileMonitor.cpp (hunk 1 above)
Comment (profnandaa) [reply LogMonitor/src/LogMonitor/LogFileMonitor.cpp]: oh, got it.

File: LogMonitor/LogMonitorTests/LogFileMonitorTests.cpp (hunk 2 above)
Comment (profnandaa) [reply LogMonitor/LogMonitorTests/LogFileMonitorTests.cpp:953]: got it, thx.

**Comment by mloskot** [comment]
Awesome! Thank you very much!

**Comment by mloskot** [comment]
Is there  a chance for LogMonitor release w/ this feature?
AFAIU, current 1.2 does not include it (release notes do not mention it)

**Comment by iankingori** [comment]
Hey @mloskot, you are correct, this is not part of  v1.2. We are currently testing the feature and intend to include it as part of our November release in around 2 weeks time.
